### PR TITLE
Reorganize packaging and hopefully fix tests

### LINKED ISSUE #269: Stop building Docker image
@mattcieslak and I were chatting in SCRUM, and we were thinking it might be easiest to stop building a Docker image for CuBIDS. If we treat CuBIDS as a standard Python library, then we just need to ensure that it checks for git, git-annex, and datalad when users want to use version control. If those checks aren't already in CuBIDS, we should be able to adapt something from BABS.

If the Docker image is a crucial element in anyone's CuBIDS workflow, please let us know.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #269.

Changes proposed:

- Remove the Docker and Singularity build/test steps from the CircleCI config.
- Replace setuptools-based packaging with hatch. This means dropping setup.py, setup.cfg, tox.ini, and requirements.txt in favor of pyproject.toml. It should also infer the correct version string, rather than relying on a hardcoded version in the code.
- Move the test files into `cubids/tests`.
- Add a CITATION.cff file with the Zenodo authors and the CuBIDS paper.
- Remove unused package*.json files.

### CODE REVIEW

**Comment by tsalo** [comment]
I'm stumped. I need to install CuBIDS in the Docker image in order to access the package's version, but I can't seem to get a base image with Docker installed within it _and_ Python 3.8+ (which is necessary to install a Python library using pyproject.toml).

**Comment by tsalo** [comment]
CircleCI has both Docker executors and machine executors. The machine executor can run Docker without having to set up a remote docker.

**Comment by tsalo** [comment]
It's working!

**Comment by tsalo** [comment]
@mattcieslak if we can tackle this PR next, it should lead into #268 pretty cleanly.

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,116 @@
...
+
+[tool.hatch.build.targets.wheel]
+packages = ["cubids"]
+exclude = [
+    "cubids/tests/data",  # Large test data directory
+]
+
+##  The following two sections configure setuptools_scm in the hatch way
+
+[tool.hatch.version]
```
Comment (mattcieslak) [inline pyproject.toml:109]: Do you know why github is rendering this in red?

**Review by mattcieslak — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (tsalo) [reply pyproject.toml:109]: No clue. It's doing it in the ASLPrep pyproject.toml as well though...

Comment (tsalo) [reply pyproject.toml:109]: I think it might be a bug on GitHub's end, since installation seems to work just fine.

**Pull request merged**